.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,27 @@
+name: Run Tests
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+    paths:
+      - 'functions/**'
+jobs:
+  run_tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build
+        run: docker-compose build --no-cache
+      - name: Run unit tests
+        run: docker run checkmate_emulator:latest npm test
+      - name: Start docker compose for testing
+        run: docker-compose up -d
+      - name: Wait for emulator to start
+        run: timeout 30 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:15001/checkmate-373101/asia-southeast1/healthcheck)" != "200" ]]; do sleep 1; done' || false
```

Comment:
`docker-compose up -d` returns successfully before containers in it fully instantiate, so we need to wait for the emulator to actually be ready. This checks for the `healthcheck` endpoint every second for 30 secs.